api/v1/registry.proto

```diff
@@ -58,6 +60,23 @@ message VersionedApplicationSpec {
     repeated Flavor flavors = 40;
 }
 
+// for use when an installation spec contains multiple steps
```

Comment:
I like this, my main concern is how does this interact with the layers potentially defined on the flavor. Presumably, the layers would apply to every installation step, but that could lead to errors if the layer can't be applied multiple times. Perhaps this only works if there are no layers, or in cases where the layers are restricted?